server/src/main/java/org/candlepin/audit/QpidConnection.java

```diff
@@ -0,0 +1,238 @@
+     * the client must use connect() method to try establish new connection
+     */
+    private Connection connection;
+    /**
+     * A Topic session is associated with connection, when the connection is
+     * recreated, so must be the TopicSession
+     */
+    private TopicSession session;
+    /**
+     * For each combination of target (Consumer, Pool, etc) and type of event
+     * (CREATED, DELETED, ...) we have a special TopicPublisher. The reason we have
+     * one publisher for every combo is that we want the events to end up in
+     * event exchange under specific routing key. The qpid jms client allows us to
+     * do this using TopicPublisher
+     */
+    private Map<Target, Map<Type, TopicPublisher>> producerMap;
+    private static Logger log = LoggerFactory.getLogger(QpidConnection.class);
+    private InitialContext ctx = null;
+    private STATUS connectionStatus = STATUS.DOWN;
+    private QpidConfigBuilder config;
+
+    /**
+     * This class is a singleton, just in case that multiple threads
+     * try to reconnect concurrently, we want to shield ourselves
+     */
+    private static Object connectionLock = new Object();
+
+    /**
+     * Status of the connection as Candlepin sees it
+     * @author fnguyen
+     *
+     */
+    public enum STATUS {
+        CONNECTED,
+        DOWN
+    }
+
+    @Inject
+    public QpidConnection(QpidConfigBuilder config) {
+        try {
+            this.config = config;
+            ctx = new InitialContext(config.buildConfigurationProperties());
+            connectionFactory = createConnectionFactory();
+        }
+        catch (NamingException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void sendTextMessage(Target target, Type type, String msg) {
+        if (connectionStatus == STATUS.DOWN) {
+            throw new RuntimeException("The Qpid connection is down, " +
+                "please use connect() to reestablish connection.");
+        }
+
+        try {
+            Map<Type, TopicPublisher> m = this.producerMap.get(target);
+            if (m != null) {
+                TopicPublisher tp = m.get(type);
+                tp.send(session.createTextMessage(msg));
+            }
+        }
+        catch (JMSException ex) {
+            log.error("Unable to send event: " + msg, ex);
+            connectionStatus = STATUS.DOWN;
+            throw new RuntimeException("Error sending event to message bus", ex);
+        }
+    }
+
+    /**
+     * This indempotent method will establish connection to qpid broker. Per JMS standard
+     * it must recreate all JMS objects such as Connection, TopicSession, TopicPublisher.
+     * @throws Exception errors during connecting to the Broker
+     */
+    public void connect() throws Exception {
+        synchronized (connectionLock) {
+            if (connectionStatus == STATUS.CONNECTED) {
+                log.info("Already connected to Qpid Broker, ignoring this connect request");
```

Comment:
I will remove this check all together. I found that its better to just let the reconnect to go through. The connectionStatus variable value is not reliable in concurrent environment.